internal/handlers/pg/msg_renamecollection.go

```diff
@@ -0,0 +1,141 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pg
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/pg/pgdb"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgRenameCollection implements HandlerInterface.
+func (h *Handler) MsgRenameCollection(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	var err error
+	dbPool, err := h.DBPool(ctx)
+
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.Unimplemented(document, "writeConcern", "comment"); err != nil {
+		return nil, err
+	}
+
+	var dropTarget bool
+
+	if dropTarget, err = common.GetOptionalParam(document, "dropTarget", false); err != nil {
+		return nil, err
+	}
+	_ = dropTarget
+
+	var qp pgdb.QueryParams
+
+	if qp.DB, err = common.GetRequiredParam[string](document, "$db"); err != nil {
+		return nil, err
+	}
+
+	if qp.DB != "admin" {
+		return nil, commonerrors.NewCommandErrorMsg(
+			commonerrors.ErrUnauthorized,
+			"renameCollection may only be run against the admin database.",
+		)
+	}
+
+	sourceNamespace, err := common.GetRequiredParam[string](document, document.Command())
+	if err != nil {
+		return nil, err
+	}
+
+	targetNamespace, err := common.GetRequiredParam[string](document, "to")
+	if err != nil {
+		return nil, err
```

Comment:
This err also needs to be tested. For example, what if the key "to" does not exist. What if the value of the key was not string.

Reply:
Yes the above is true.

```
test> db.runCommand({renameCollection: "test.foo", })
MongoServerError: required parameter "to" is missing
```

Reply:
Oops my mistake, done. I hope. see [here](https://github.com/FerretDB/FerretDB/pull/2343/files#diff-cf8e6974b8dde0f1aa07175594275a3f09d0bb9f2b196a8f45f8071e53dd0e1eR851-R859).

Also, I cannot see a way to test for non-string types because it complains about type assertion, I even tried this:

```go
			var to any
			switch tc.to.(type) {
			case string:
				to = tc.to.(string)
			}
			cmd := bson.D{
				{"renameCollection", sourceNamespace},
				{to, targetNamespace},
			}
```